### ISSUE TITLE
The Mellanox nic  does not work in the latest branch and works fine on 1.21.1-tag

### ISSUE THREAD

**Issue by IAseven** [ipxe/ipxe#1222]
Mellanox nic:
![图片](https://github.com/ipxe/ipxe/assets/38178039/76a746cc-b90e-4cb1-ac78-7325c3f016e0)

The network adapter is started and the ip address is configured properly, but the gateway cannot be pinged from the master-branch，

What data do I need to check to solve this bug,

Thank you very much for your answers

**Comment by NiKiZe** [comment]
This is probably a duplicate of existing issues.
But to verify which binary are you using, and what is the output of ifstat in ipxe?

**Comment by IAseven** [comment]
@NiKiZe 
I'm very sorry, this is my problem, I used the latest version wrong, I re-downloaded the latest code and can communicate normally after compilation.
I am very sorry that I did not inquire relevant questions in the issues in advance, which has delayed your precious time.
The next time I have a problem, I look at the issues first
Finally, thank you very much for your selfless dedication to ipxe over the past ten years.
